airbyte-integrations/connectors/source-exchange-rates/source_exchange_rates/source.py

```diff
@@ -110,7 +110,17 @@ def check_connection(self, logger, config) -> Tuple[bool, any]:
             logger.info(f"Ping response code: {status}")
             if status == 200:
                 return True, None
-            return False, resp.text
+            # When API requests is sent but the requested data is not available or the API call fails
+            # for some reason, a JSON error is returned.
+            # https://exchangeratesapi.io/documentation/#errors
+            error = resp.json().get("error")
+            code = error.get("code")
+            message = error.get("message")
```

Comment:
In link you put in a comment there is no `message` field in the response body. There is an `info` field. Is this your typo or they can send both `message` and `info` fields? If yes then we need to add support for `info` also.

Reply:
I checked this, API usual returns message field. But I will add handling of info field just in case.